File: mvp/sdk/typescript/src/api/agents.ts
import { BaseAPI } from "./base";
import {
  AgentRun,
  AgentRunRequest,
  AgentReactRequest,
  PaginationParams,
} from "../types";

/**
 * Agents API — autonomous AI agents with planning and 65+ available actions.
 */
export class AgentAPI extends BaseAPI {
  /** Execute an autonomous AI agent with a goal. */
  async run(data: AgentRunRequest): Promise<AgentRun> {
    return this._post<AgentRun>("/api/agents/run", data);
  }

  /** Run a ReAct agent (reason + act loop). */
  async react(data: AgentReactRequest): Promise<AgentRun> {
    return this._post<AgentRun>("/api/agents/react", data);
  }

  /** List past agent runs. */
  async listRuns(params?: PaginationParams): Promise<AgentRun[]> {
    return this._get<AgentRun[]>("/api/agents/runs", {
      limit: params?.limit,
      offset: params?.offset,
    });
  }

  /** Get agent run details with step history. */
  async getRun(id: string): Promise<AgentRun> {
    return this._get<AgentRun>(`/api/agents/runs/${id}`);
  }

  /** Cancel a running agent. */
  async cancelRun(id: string): Promise<{ status: string }> {
    return this._post<{ status: string }>(`/api/agents/runs/${id}/cancel`);
  }
}

File: mvp/sdk/python/saas_ia/api/agents.py
"""Agents API — autonomous AI agents with planning and 65+ actions."""

from __future__ import annotations

from typing import Any

from saas_ia.api.base import BaseAPI


class AgentAPI(BaseAPI):
    """Wraps ``/api/agents`` endpoints."""

    async def run(
        self,
        goal: str,
        *,
        context: str | None = None,
        max_steps: int | None = None,
    ) -> dict[str, Any]:
        """Execute an autonomous AI agent with a goal."""
        body: dict[str, Any] = {"goal": goal}
        if context:
            body["context"] = context
        if max_steps:
            body["max_steps"] = max_steps
        return await self._post("/api/agents/run", json=body)

    async def react(
        self,
        instruction: str,
        *,
        context: str | None = None,
        max_iterations: int | None = None,
    ) -> dict[str, Any]:
        """Run a ReAct agent (reason + act loop)."""
        body: dict[str, Any] = {"instruction": instruction}
        if context:
            body["context"] = context
        if max_iterations:
            body["max_iterations"] = max_iterations
        return await self._post("/api/agents/react", json=body)

    async def list_runs(
        self,
        *,
        limit: int | None = None,
        offset: int | None = None,
    ) -> list[dict[str, Any]]:
        """List past agent runs."""
        return await self._get(
            "/api/agents/runs",
            params={"limit": limit, "offset": offset},
        )

    async def get_run(self, run_id: str) -> dict[str, Any]:
        """Get agent run details with step history."""
        return await self._get(f"/api/agents/runs/{run_id}")

    async def cancel_run(self, run_id: str) -> dict[str, Any]:
        """Cancel a running agent."""
        return await self._post(f"/api/agents/runs/{run_id}/cancel")
